src/karthik/group-people-data/groupPeopleData.ts

```diff
@@ -0,0 +1,44 @@
+interface People {
+  _id: string;
+  name: string;
+  gender: string;
+  email: string;
+  phone: string;
+  city: string;
+}
+
+interface GroupOfPeopleByCity {
+  count: number;
+  city: string;
+  people: Array<People>;
+}
+
+const groupPeopleByCity = (
+  arrayOfObject: Array<any>
+): Array<GroupOfPeopleByCity> => {
+  if (arrayOfObject === []) return [];
+  const grouped = {};
+  const property = 'city';
+  arrayOfObject.map((obj) => {
+    if (obj[property]) {
+      if (grouped[obj[property]]) {
+        grouped[obj[property]].push(obj);
+      } else {
+        grouped[obj[property]] = [obj];
+      }
+    }
+  });
+
+  const res: Array<GroupOfPeopleByCity> = [];
+  for (let str in grouped) {
```

Comment:
Not an issue, Here to iterate over object keys we can use `map` on object keys and here we will get the result array length equal to number of keys in the grouped object.
```
const res = Object.keys(grouped).map(key => ({ city: key, people: grouped[key]}));
```

Reply:
fixed.